File: removeElement.py
from typing import List
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        count = 0  # Initialize count for elements not equal to val
        
        # Iterate through the array
        for num in nums:
            # If the current element is not equal to val
            if num != val:
                # Keep the element and update the count
                nums[count] = num
                count += 1
        
        # Return the count of elements not equal to val
        return count

# Test the function
if __name__ == "__main__":
    solution = Solution()
    nums = [3, 2, 2, 3]
    val = 3
    k = solution.removeElement(nums, val)
    print("Number of elements in nums which are not equal to val:", k)
    print("Modified nums array:", nums[:k])

File: Week1/removeElement.java
package Week1;

public class removeElement {
    public int RemoveElement(int[] nums, int val) {
        int count = 0; // Initialize count for elements not equal to val
        
        // Iterate through the array
        for (int i = 0; i < nums.length; i++) {
            // If the current element is not equal to val
            if (nums[i] != val) {
                // Keep the element and update the count
                nums[count] = nums[i];
                count++;
            }
        }
        
        // Return the count of elements not equal to val
        return count;
    }

    public static void main(String[] args) {
        removeElement solution = new removeElement();
        int[] nums = {4,9,7,4};
        int val = 4;
        int k = solution.RemoveElement(nums, val);
        System.out.println("Number of elements in nums which are not equal to val: " + k);
        System.out.print("Modified nums array: ");
        for (int i = 0; i < k; i++) {
            System.out.print(nums[i] + " ");
        }
    }
}

